sympy/solvers/tests/test_solveset.py

```diff
@@ -178,6 +178,14 @@ def test_issue_11536():
     assert solveset(0**x - 1, x, S.Reals) == FiniteSet(0)
 
 
+def test_issue_17882():
+    equation = solveset(-8*x**2/(9*(x**2 - 1)**(4/3)) + 4/(3*(x**2 - 1)**(1/3)), x, S.Complexes)
+    equation = str(equation)
```

Comment:
We should avoid using `str` for testing.

Reply:
Should I implement an alternative approach??

Reply:
Use rational numbers for the coefficients, like `S(4)/3`. You should be able to compare the solution exactly.

Reply:
Can you rewrite the equation in the required way, so that I get a better idea?

Reply:
Specifically, you should write the equation like
`solveset(-8*x**2/(9*(x**2 - 1)**(S(4)/3)) + 4/(3*(x**2 - 1)**(S(1)/3)), x, S.Complexes)`

Reply:
I will change this immediately. Anything else which needs to be changed?

Reply:
I am getting exception on this testcase when I make the above-mentioned changes.

Reply:
What kind of exception are you facing? Have you changed ‘solution’ to use rational numbers?

Reply:
> Specifically, you should write the equation like
> `solveset(-8*x**2/(9*(x**2 - 1)**(S(4)/3)) + 4/(3*(x**2 - 1)**(S(1)/3)), x, S.Complexes)`

solveset is not able to solve this equation when rational numbers are introduced.

Reply:
If this is an error, I can open up an issue regarding this, or shall we try to resolve the issue using this PR itself.

Reply:
It looks like the issue is not resolved 'well'.
If you just want to stop investigating further and get over with the unevaluated result like 
`ConditionSet(x, Eq(-8x**2(x2 - 1)(-1.33333333333333)/9 + 4*(x2 - 1)(-0.333333333333333)/3, 0), S.Complexes)`, I would merge this soon.

But I would rather update the original post and keep the thread open, and make sure that it should resolve 
1. Some logic that leads to infinite recursions
2. It should actually 'solve' the equation rather than returning conditionset.

Reply:
Actually, it looks like is not resolving anything because your original test case 
`solveset(-8*x**2/(9*(x**2 - 1)**(4/3)) + 4/(3*(x**2 - 1)**(1/3)), x, S.Complexes)`
works as same in the master.

It looks like you should actually get
`solveset(-8*x**2/(9*(x**2 - 1)**(S(4)/3)) + 4/(3*(x**2 - 1)**(S(1)/3)), x, S.Complexes)`
to work

Reply:
Ok, no issue.
But in my opinion, we can **merge this and keep the thread open for further improvements**.

Reply:
No, unfortunately for the reason in https://github.com/sympy/sympy/pull/18324#discussion_r366727879

Reply:
ok, i didn't observe this, i will start working on this immediately.